### PR TITLE
[cudagraphs] add more info to skip messages

### PR DESCRIPTION
**Pull request by xmfan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #124804
* __->__ #124700
* #119729

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

File: torch/_inductor/cudagraph_utils.py [hunk 1]
```diff
@@ -90,7 +90,13 @@ def check_for_mutation(
 
     else:
         has_mutation = len(func.mutated_input_idxs) != 0
-        return None if not has_mutation else default_msg
+        return (
+            None
+            if not has_mutation
+            else format_default_skip_message(
+                f"mutated inputs ({len(func.mutated_input_idxs)} instances)"
+            )
```
Comment (eellison) [inline torch/_inductor/cudagraph_utils.py:98]: https://github.com/pytorch/pytorch/blob/main/test/inductor/test_cudagraph_trees.py#L255-L257

```
            FileCheck().check(
                "skipping cudagraphs due to mutation on input. Found from"
            ).check("torch.logical_xor").run(captured_output[0])
```

Are we not printing mutation traceback here ? where does this even gets printed

**Review by eellison — COMMENTED** [review]
looks good, thanks for changes! would you mind adding a couple tests

**Comment by eellison** [comment]
looks like there are a couple test failures tho

File: torch/_inductor/cudagraph_utils.py (hunk 1 above)
Comment (xmfan) [reply torch/_inductor/cudagraph_utils.py:98]: probably from the `if torch._inductor.config.triton.cudagraph_trees:` block on line 75

**Review by eellison — APPROVED** [review]
😍 😍 😍 

you might need to rebase @BoyuanFeng just touched this code as well

**Pull request closed without merging**